### ISSUE TITLE
Font names don't match across macOS and Windows

### ISSUE THREAD

**Issue by PrathamTharwani** [libass/libass#533]
[bugreport.zip](https://github.com/libass/libass/files/6873635/bugreport.zip)
Above zip file contains a font file 'NEGOTIATEFREE.TTF' which was muxed into the video along with the included subtitle file.

Subtitle file was created on a macOS system and specifies the font name as 'Negotiate' (as seen in the Lyrics styles). Windows reports the same font as 'Negotiate Free' (seen in Windows font installer) and as such when the video is played in mpv, it fails to match the font and falls back to the default. On editing the subtitle file to say 'Negotiate Free' instead of 'Negotiate', playback proceeds normally and with the expected font. Issue has been observed with at least 3 other fonts so far.

**Comment by TheOneric** [comment]
This seems to be a broken font file, similar to, but not quite as bad as the original SVN-Student issue in #315.
FontForge warns me that the “Mac  string is a subset of the Windows string in the 'name' table for the Family string in the English (US) language” and also the “Styles (SubFamily)” differ with “Free” for Mac and “Regular” for Windows.
fontconfig *(both `fc-match` and our fontconfig-font-provider)* matches both `Negotiate` and `Negotiate Free` to the provided font.


Ideally, the font file should be fixed to have consistent naming, but since we know VSFilters will always prefer the Microsoft names *(and unfortunately fonts with inconsistent names are not as rare as one would hope)*, we probably can and want to do something about this on libass' side anyway.

For reference, using fontconfig 2.13.1 here's what the `fonts-debug` branch prints with `Negotiate Free` in the ASS:
```
libass API version: 0x1501000
libass source: commit: 0.15.1-54-g51bf6e8ad8fb71cc4e616be28a7f49ff422a2c98
Raster: FreeType 2.10.4
Shaper: FriBidi 1.0.8 (SIMPLE) HarfBuzz-ng 2.7.4 (COMPLEX)
Initialized
new font:
  family: 'Negotiate'
  family: 'Negotiate Free'
  fullname: 'Negotiate Free'
  slant: 0
  weight: 400
  width: 100
  postscript_name: NegotiateFree
new font:
  family: 'SVN-Student'
  fullname: 'SVN-Student'
  fullname: 'UTM Ong Do Gia'
  slant: 0
  weight: 400
  width: 100
  postscript_name: SVN-Student    [ <--- Interestingly this differs from what fontforge says: It complains about the postscript name "UTM Ong Do Gia" being invalid]

[Removed logs about unrelated fonts]

Using font provider fontconfig
File size: 15431
[0x5627eb290200] Style: Default,Arial,48,&H00FFFFFF,&H000000FF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,2,2,2,10,10,10,1
[0x5627eb290200] Style: Lyrics-Jap,Negotiate Free,32,&H00FBFBFB,&H000000FF,&H0035050D,&HAF000000,0,0,0,0,100,100,0,0,1,1.2,0,1,20,20,11,1
[0x5627eb290200] Style: Lyrics-Eng,Negotiate Free,32,&H00FBFBFB,&H000000FF,&H0035050D,&HAF000000,0,0,0,0,100,100,0,0,1,1.2,0,1,20,20,11,1
[0x5627eb290200] Style: Lyrics-Bg,Negotiate Free,32,&H00FBFBFB,&H000000FF,&H0035050D,&HAF000000,0,0,0,0,100,100,0,0,1,1.2,0,1,20,20,11,1
Added subtitle file: '~/TEST/533_reduced-sample.ass' (5 styles, 116 events)
add_face(ch=U+0000): font->n_faces=0
ass_font_select(code=U+0000): family=[Negotiate Free], bold=400, italic=0
select_font(code=U+0000, family=[Negotiate Free], bold=400, italic=0)
n_fullname=1, fullnames:
    Negotiate Free
find_font returned path [~/.fonts/NEGOTIATEFREE.TTF], index 0, PostScript name [NegotiateFree]
first select_font returned ~/.fonts/NEGOTIATEFREE.TTF
fontselect: (Negotiate Free, 400, 0) -> ~/.fonts/NEGOTIATEFREE.TTF, 0, NegotiateFree
ass_font_select returned ~/.fonts/NEGOTIATEFREE.TTF
```
With `Negotiate` in the ASS it's almost the same, just without `Free` at the end::
```
[…]

Added subtitle file: '~/TEST/533_reduced-sample.ass' (5 styles, 116 events)
add_face(ch=U+0000): font->n_faces=0
ass_font_select(code=U+0000): family=[Negotiate], bold=400, italic=0
select_font(code=U+0000, family=[Negotiate], bold=400, italic=0)
n_fullname=1, fullnames:
    Negotiate
find_font returned path [~/.fonts/NEGOTIATEFREE.TTF], index 0, PostScript name [NegotiateFree]
first select_font returned ~/.fonts/NEGOTIATEFREE.TTF
fontselect: (Negotiate, 400, 0) -> ~/.fonts/NEGOTIATEFREE.TTF, 0, NegotiateFree
ass_font_select returned ~/.fonts/NEGOTIATEFREE.TTF
```

**Comment by PrathamTharwani** [comment]
Little bit of clarification:
The video I'm trying to play was not created by me, but was created by someone else on a mac sometime around 2017.

To reiterate I am on Windows 10 with:
```
libass API version: 0x1501000
libass source: commit: 0.15.1-41-g5733e1c28b92ebbf090bea8f25e252aa40e0b9c4
```
as taken from the mpv logs.
And the behavior on my machine is that ```Negotiate Free``` in the ASS works but ```Negotiate``` (which is how I found it when I extracted the subs from the video file) does not work.

Now since I do not have access to a mac I talked to the person who originally made the file and he reports that on a newer version of mpv and libass  ```Negotiate``` no longer works but ```Negotiate Free``` does work.

So if I understand this situation correctly, current behavior is that ```Negotiate``` does not work on either OS and ```Negotiate Free``` does
